cassovary-core/src/main/scala/com/twitter/cassovary/util/SortedSeq.scala

```diff
@@ -0,0 +1,114 @@
+package com.twitter.cassovary.util
+
+import java.util
+
+import scala.collection.mutable
+import scala.reflect.ClassTag
+
+/**
+ * Sequence of sorted elements.
+ * 
+ * Assumes each element may appear in the Seq at most once.
+ */
+trait StrictlySortedSeqLike[T] extends mutable.IndexedSeq[T] {
```

Comment:
This trait sounds like SortedSet. Use that instead adding the 'intersect' method?